ManPages/tsManPagesLibGUI/tsWxChoice.man

Help on module tsWxChoice:

NAME
    tsWxChoice

FILE
    ./tsLibGUI/tsWxPkg/src/tsWxChoice.py

DESCRIPTION
    tsWxChoice.py - Class to emulate the wxPython API for non-graphical,
    curses-based platforms.

CLASSES
    tsWxControlWithItems.ControlWithItems(tsWxControl.Control, tsWxItemContainer.ItemContainer)
        Choice
    
    class Choice(tsWxControlWithItems.ControlWithItems)
     |  A Choice control is used to select one of a list of strings. Unlike a
     |  wx.ListBox, only the selection is visible until the user pulls down
     |  the menu of choices.
     |  
     |  Events
     |  
     |  EVT_CHOICE  Sent when an item in the list is selected.
     |  
     |  Method resolution order:
     |      Choice
     |      tsWxControlWithItems.ControlWithItems
     |      tsWxControl.Control
     |      tsWxWindow.Window
     |      tsWxEvtHandler.EvtHandler
     |      tsWxObject.Object
     |      tsWxItemContainer.ItemContainer
     |      __builtin__.object
     |  
     |  Methods defined here:
     |  
     |  Create(self, parent, id=-1, pos=(-1, -1), size=(-1, -1), choices=[], style=0, validator=None, name='choice', pixels=True)
     |      Actually create the GUI Choice control for 2-phase creation
     |  
     |  GetCurrentSelection(self)
     |      Unlike GetSelection which only returns the accepted selection value,
     |      i.e. the selection in the control once the user closes the dropdown
     |      list, this function returns the current selection. That is, while the
     |      dropdown list is shown, it returns the currently selected item in it.
     |      When it is not shown, its result is the same as for the other function.
     |  
     |  __init__(self, parent, id=-1, pos=(-1, -1), size=(-1, -1), choices=[], style=0, validator=None, name='choice')
     |      Create and show a Choice control
     |  
     |  ----------------------------------------------------------------------
     |  Static methods defined here:
     |  
     |  GetClassDefaultAttributes(variant=0)
     |      Get the default attributes for this class. This is useful if you want
     |      to use the same font or colour in your own control as in a standard
     |      control -- which is a much better idea than hard coding specific
     |      colours or fonts which might look completely out of place on the
     |      user system, especially if it uses themes.
     |      
     |      The variant parameter is only relevant under Mac currently and is
     |      ignored under other platforms. Under Mac, it will change the size of
     |      the returned font. See wx.Window.SetWindowVariant for more about this.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors defined here:
     |  
     |  CurrentSelection
     |      Unlike GetSelection which only returns the accepted selection value,
     |      i.e. the selection in the control once the user closes the dropdown
     |      list, this function returns the current selection. That is, while the
     |      dropdown list is shown, it returns the currently selected item in it.
     |      When it is not shown, its result is the same as for the other function.
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxControl.Control:
     |  
     |  Command(self, event)
     |      Simulates the effect of the user issuing a command to the item.
     |  
     |  GetAlignment(self)
     |      Get the control alignment (left/right/centre, top/bottom/centre)
     |  
     |  GetLabelText(self)
     |      Get just the text of the label, without mnemonic characters ("&")
     |  
     |  tsUpdate(self)
     |      Draw the actual features of the Control.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from tsWxControl.Control:
     |  
     |  Alignment
     |      Get the control alignment (left/right/centre, top/bottom/centre)
     |  
     |  LabelText
     |      Get just the text of the label, without mnemonic characters ("&")
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxWindow.Window:
     |  
     |  AcceptsFocus(self)
     |      This method may be overridden in the derived classes to return false
     |      to indicate that this control does not accept input at all
     |  
     |  AcceptsFocusFromKeyboard(self)
     |      This method may be overridden in the derived classes to return false
     |      to indicate that while this control can, in principle, have focus if
     |      the user clicks it with the mouse, it should not be included in the
     |      TAB traversal chain when using the keyboard.
     |  
     |  AcceptsFocusRecursively(self)
     |      Overridden to indicate whether this window or one of its children
     |      accepts focus.
     |  
     |  AddChild(self, child)
     |      Adds a child window. This is called automatically by window creation
     |      functions so should not be required by the application programmer.
     |  
     |  AddPendingEvent(self, event)
     |      See ProcessEvent() for more info about why you should not use
     |      this function and the reason for making this function protected
     |      in wxWindow.
     |      
     |      Reimplemented from wxEvtHandler.
     |  
     |  AdjustForLayoutDirection(self, x, width, widthTotal)
     |      Mirror coordinates for RTL layout if this window uses it and
     |      if the mirroring is not done automatically like Win32.
     |  
     |  AdjustForParentClientOriginXY(self, x, y, sizeFlags)
     |  
     |  AlwaysShowScrollbars(self)
     |      Call this function to force one or both scrollbars to be always
     |      shown, even if the window is big enough to show its entire contents
     |      without scrolling.
     |  
     |  AssociateHandle(self, handle)
     |      Associate the window with a new native handle.
     |  
     |  CacheBestSize(self, size)
     |      Sets the cached best size value.
     |  
     |  CanScroll(self, orient)
     |      Returns True if this window can have a scroll bar in this
     |      orientation.
     |  
     |  CanSetTransparent(self)
     |      Returns true if the system supports transparent windows and
     |      calling SetTransparent() may succeed.
     |      
     |      If this function returns false, transparent windows are definitely
     |      not supported by the current system.
     |      
     |      Reimplemented in wxTopLevelWindow.
     |      
     |      NOTE: On X-windows systems the X server must have the composite
     |      extension loaded, and there must be a composite manager program
     |      (such as xcompmgr) running.
     |  
     |  CaptureMouse(self)
     |      Directs all mouse input to this window.
     |      
     |      Call ReleaseMouse() to release the capture.
     |      
     |      Note that wxWindows maintains the stack of windows having captured
     |      the mouse and when the mouse is released the capture returns to the
     |      window which had had captured it previously and it is only really
     |      released if there were no previous window. In particular, this means
     |      that you must release the mouse as many times as you capture it,
     |      unless the window receives the wx.MouseCaptureLostEvent event.
     |      
     |      Any application which captures the mouse in the beginning of some
     |      operation must handle wx.MouseCaptureLostEvent and cancel this
     |      operation when it receives the event. The event handler must not
     |      recapture mouse.
     |  
     |  Center(self, direction=12)
     |      Centers the window. The parameter specifies the direction for
     |      cetering, and may be wx.HORIZONTAL, wx.VERTICAL or wx.BOTH. It may
     |      also include wx.CENTER_ON_SCREEN flag if you want to center the
     |      window on the entire screen and not on its parent window. If it is
     |      a top-level window and has no parent then it will always be centered
     |      relative to the screen.
     |  
     |  CenterOnParent(self, dir=12)
     |      Center with respect to the the parent window.
     |  
     |  CenterOnScreen(self, direction=12)
     |      Center the window on screen.
     |  
     |  Centre(self, direction=12)
     |      Centers the window. The parameter specifies the direction for
     |      cetering, and may be wx.HORIZONTAL, wx.VERTICAL or wx.BOTH. It may
     |      also include wx.CENTER_ON_SCREEN flag if you want to center the
     |      window on the entire screen and not on its parent window. If it is
     |      a top-level window and has no parent then it will always be centered
     |      relative to the screen.
     |  
     |  CentreOnParent(self, dir=12)
     |      Center with respect to the the parent window.
     |  
     |  CentreOnScreen(self, direction=12)
     |      Center the window on screen.
     |  
     |  ClearBackground(self)
     |      Clears the window by filling it with the current background colour.
     |      
     |      Does not cause an erase background event to be generated.
     |      
     |      Notice that this uses wxClientDC to draw on the window and the
     |      results of doing it while also drawing on wxPaintDC for this
     |      window are undefined. Hence this method should not be used
     |      from EVT_PAINT handlers, just use wxDC::Clear() on the wxPaintDC
     |      you already use there instead.
     |  
     |  ClientToScreen(self, pt)
     |      Converts to screen coordinates from coordinates relative to this
     |      window.
     |  
     |  ClientToScreenXY(self, x, y)
     |      Converts to screen coordinates from coordinates relative to this
     |      window.
     |  
     |  ClientToWindowSize(self, size)
     |      Converts client area size size to corresponding window size.
     |  
     |  Close(self, force=True)
     |      This function simply generates a wxCloseEvent whose handler usually
     |      tries to close the window.
     |      
     |      It does not close the window itself, however.
     |      
     |      Parameters:
     |      force   false if the windows close handler should be able to veto
     |      the destruction of this window, true if it cannot.
     |      
     |      Remarks:
     |      Close calls the close handler for the window, providing an
     |      opportunity for the window to choose whether to destroy the window.
     |      Usually it is only used with the top level windows (wxFrame and
     |      wxDialog classes) as the others are not supposed to have any special
     |      OnClose() logic. The close handler should check whether the window
     |      is being deleted forcibly, using wxCloseEvent::CanVeto, in which
     |      case it should destroy the window using wxWindow::Destroy. Note
     |      that calling Close does not guarantee that the window will be
     |      destroyed; but it provides a way to simulate a manual close of
     |      a window, which may or may not be implemented by destroying the
     |      window. The default implementation of wxDialog::OnCloseWindow
     |      does not necessarily delete the dialog, since it will simply
     |      simulate an wxID_CANCEL event which is handled by the appropriate
     |      button event handler and may do anything at all. To guarantee
     |      that the window will be destroyed, call wxWindow::Destroy instead.
     |  
     |  ConvertDialogPixels(self, pt)
     |      Converts a point or size from dialog units to pixels.
     |      
     |      For the x dimension, the dialog units are multiplied by the
     |      fixed point character width. For the y dimension, the dialog
     |      units are multiplied by the fixed point character height.
     |      
     |      Remarks:
     |      Dialog units are used for maintaining a dialogs proportions
     |      even if the font changes. You can also use these functions
     |      programmatically. A convenience macro is defined:
     |      define wxDLG_UNIT(parent, pt) parent->ConvertDialogToPixels(pt)
     |  
     |  ConvertDialogPointToPixels(self, pt)
     |      Converts a point or size from dialog units to pixels.
     |  
     |  ConvertDialogSizeToPixels(self, sz)
     |      Converts a point or size from dialog units to pixels.
     |  
     |  ConvertPixelPointToDialog(self, pt)
     |      Converts a point or size from pixel units to dialog units.
     |  
     |  ConvertPixelSizeToDialog(self, sz)
     |      Converts a point or size from pixel units to dialog units.
     |  
     |  ConvertPixelsToDialog(self, pt)
     |      Converts a point or size from pixels to dialog units.
     |      
     |      For the x dimension, the pixels are divided by the fixed point
     |      character width. For the y dimension, the pixels are divided
     |      by the fixed point character height.
     |      
     |      Remarks:
     |      Dialog units are used for maintaining a dialogs proportions even
     |      if the font changes.
     |  
     |  DLG_PNT(self, pt)
     |      Converts a point or size from dialog units to pixels. Dialog units
     |      are used for maintaining a dialog proportions even if the font
     |      changes. For the x dimension, the dialog units are multiplied by the
     |      average character width and then divided by 4. For the y dimension,
     |      the dialog units are multiplied by the average character height and
     |      then divided by 8.
     |  
     |  DLG_SZE(self, sz)
     |      Converts a point or size from dialog units to pixels. Dialog units
     |      are used for maintaining a dialog proportions even if the font
     |      changes. For the x dimension, the dialog units are multiplied by the
     |      average character width and then divided by 4. For the y dimension,
     |      the dialog units are multiplied by the average character height and
     |      then divided by 8.
     |  
     |  Destroy(self)
     |      Destroys the window safely.
     |      
     |      Use this function instead of the delete operator, since different
     |      window classes can be destroyed differently. Frames and dialogs
     |      are not destroyed immediately when this function is called -- they
     |      are added to a list of windows to be deleted on idle time, when
     |      all the windows events have been processed. This prevents problems
     |      with events being sent to non-existent windows.
     |      
     |      Returns:
     |      true if the window has either been successfully deleted, or it has
     |      been added to the list of windows pending real deletion.
     |  
     |  DestroyChildren(self)
     |      Destroys all children of a window.
     |  
     |  Disable(self)
     |      Disables the window for user input, same as Enable(False).
     |  
     |  DissociateHandle(self)
     |      Dissociate the current native handle from the window.
     |  
     |  DoCaptureMouse(self)
     |  
     |  DoCentre(self, dir)
     |      Centres the window.
     |      
     |      Parameters:
     |      direction  Specifies the direction for the centring. May be
     |      wxHORIZONTAL, wxVERTICAL or wxBOTH. It may also include the
     |      wxCENTRE_ON_SCREEN flag.
     |      
     |      Remarks:
     |      This function is not meant to be called directly by user code,
     |      but via Centre, Center, CentreOnParent, or CenterOnParent. This
     |      function can be overriden to fine-tune centring behaviour.
     |  
     |  DoFindFocus(self)
     |      Finds the window or control which currently has the keyboard focus.
     |      
     |      Remarks:
     |      
     |      Note that this should have been a static function, so it can be
     |      called without needing a wxWindow pointer.
     |  
     |  DoGetBestClientSize(self)
     |  
     |  DoGetBestSize(self)
     |      Gets the size which best suits the window: for a control, it
     |      would be the minimal size which does not truncate the control,
     |      for a panel - the same size as it would have after a call to Fit().
     |      The default implementation of this function is designed for use
     |      in container windows, such as wxPanel, and works something like
     |      this:
     |      
     |      1) If the window has a sizer then it is used to calculate the
     |      best size.
     |      
     |      2) Otherwise if the window has layout constraints then those are
     |      used to calculate the best size.
     |      
     |      3) Otherwise if the window has children then the best size is set
     |      to be large enough to show all the children.
     |      
     |      4) Otherwise if there are no children then the windows minimal
     |      size will be used as its best size.
     |      
     |      5) Otherwise if there is no minimal size set, then the current
     |      size is used for the best size.
     |  
     |  DoGetBorderSize(self)
     |      Return the thickness (width and height) of the display
     |      occupied by the top, bottom, left and right border
     |  
     |  DoGetScreenPosition(self, x, y)
     |  
     |  DoGetSibling(self, order)
     |      Return the sibling of the window that is in the specified order.
     |  
     |  DoGetVirtualSize(self)
     |  
     |  DoPhase(self, phase)
     |      Do a phase of evaluating child constraints
     |  
     |  DoReleaseMouse(self)
     |  
     |  DoSetSizeHints(self, minW, minH, maxW, maxH, incW=None, incH=None)
     |  
     |  DoSetVirtualSize(self, x, y)
     |  
     |  DoSetWindowVariant(self, variant=None)
     |      Adjust the font height to correspond to our new variant (notice that
     |      we are only called if something really changed)
     |  
     |  DoUpdateWindowUI(self, event)
     |      Does the window-specific updating after processing the update event.
     |      
     |      This function is called by UpdateWindowUI() in order to check
     |      return values in the wxUpdateUIEvent and act appropriately.
     |      For example, to allow frame and dialog title updating, wxWidgets
     |      implements this function as follows:
     |      
     |      // do the window-specific processing after processing the update event
     |      void wxTopLevelWindowBase::DoUpdateWindowUI(wxUpdateUIEvent& event)
     |      {
     |          if ( event.GetSetEnabled() )
     |              Enable(event.GetEnabled());
     |      
     |          if ( event.GetSetText() )
     |          {
     |              if ( event.GetText() != GetTitle() )
     |                  SetTitle(event.GetText());
     |          }
     |      }
     |  
     |  DragAcceptFiles(self, accept)
     |      Enables or disables eligibility for drop file events (OnDropFiles).
     |      
     |      Parameters:
     |      accept  If true, the window is eligible for drop file events. If
     |      false, the window will not accept drop file events.
     |      
     |      Remarks:
     |      Windows only until version 2.8.9, available on all platforms since
     |      2.8.10. Cannot be used together with SetDropTarget() on non-Windows
     |      platforms.
     |  
     |  Enable(self, enable=True)
     |      Enable or disable the window for user input.
     |      
     |      Note that when a parent window is disabled, all of its children
     |      are disabled as well and they are reenabled again when the parent is.
     |      
     |      Parameters:
     |      enable  If true, enables the window for input. If false, disables
     |      the window.
     |      
     |      Returns:
     |      Returns true if the window has been enabled or disabled, false if
     |      nothing was done, i.e. if the window had already been in the
     |      specified state.
     |  
     |  FindFocus(self)
     |      Finds the window or control which currently has the keyboard focus.
     |      
     |      Remarks:
     |      
     |      Note that this should have been a static function, so it can be
     |      called without needing a wxWindow pointer.
     |  
     |  FindWindowById(self, winid, parent=None)
     |      Find the first window with the given id.
     |      
     |      If parent is NULL, the search will start from all top-level frames
     |      and dialog boxes; if non-NULL, the search will be limited to the
     |      given window hierarchy. The search is recursive in both cases.
     |  
     |  FindWindowByLabel(self, label, parent=None)
     |      Find a window by its label.
     |      
     |      Depending on the type of window, the label may be a window title
     |      or panel item label. If parent is NULL, the search will start from
     |      all top-level frames and dialog boxes; if non-NULL, the search will
     |      be limited to the given window hierarchy. The search is recursive
     |      in both cases.
     |  
     |  FindWindowByName(self, name, parent=None)
     |      Find a window by its name (as given in a window constructor or
     |      Create() function call).
     |      
     |      If parent is NULL, the search will start from all top-level frames
     |      and dialog boxes; if non-NULL, the search will be limited to the
     |      given window hierarchy.
     |      
     |      The search is recursive in both cases. If no window with such name
     |      is found, FindWindowByLabel() is called.
     |  
     |  Fit(self)
     |      Sizes the window so that it fits around its subwindows. This function
     |      will not do anything if there are no subwindows and will only really
     |      work correctly if sizers are used for the subwindows layout.
     |      
     |      Also, if the window has exactly one subwindow it is better (faster
     |      and the result is more precise as Fit adds some margin to account for
     |      fuzziness of its calculations) to call
     |      window.SetClientSize(child.GetSize()) instead of calling Fit.
     |  
     |  FitInside(self)
     |      Similar to Fit(), but sizes the interior (virtual) size of a
     |      window.
     |      
     |      Mainly useful with scrolled windows to reset scrollbars after
     |      sizing changes that do not trigger a size event, and/or scrolled
     |      windows without an interior sizer. This function similarly will
     |      not do anything if there are no subwindows.
     |  
     |  Freeze(self)
     |      Freezes the window or, in other words, prevents any updates from
     |      taking place on screen, the window is not redrawn at all.

Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-2x/tsWxGTUI_Py2x/tsDemoArchive/tsTutorialDeveloperSandbox2x/tsLibGUI/tsWxPkg/src/tsWxChoice.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  6:27:32 AM rsg>"
'''
tsWxChoice.py - Class to emulate the wxPython API for non-graphical,
curses-based platforms.
'''
#################################################################
#
# File: tsWxChoice.py
#
# Purpose:
#
#    Class to emulate the wxPython API for non-graphical,
#    curses-based platforms.
#
# Usage (example):
#
#    # Import
#
#    from tsWxChoice import Choice as wxChoice
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#    None.
#
# ToDo:
#
#    None.
#
#################################################################

__title__     = 'tsWxChoice'
__version__   = '0.2.0'
__date__      = '04/01/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

import tsWxGlobals as wx
from tsWxControl import Control
from tsWxPoint import Point as wxPoint
from tsWxRect import Rect as wxRect
from tsWxSize import Size as wxSize
from tsWxControlWithItems import ControlWithItems

#---------------------------------------------------------------------------

##DEBUG = True
DEBUG = wx.Debug_GUI_Launch | \
        wx.Debug_GUI_Progress | \
        wx.Debug_GUI_Termination | \
        wx.Debug_GUI_Exceptions

##VERBOSE = True
VERBOSE = wx.Debug_GUI_Configuration

#---------------------------------------------------------------------------

class Choice(ControlWithItems):
    '''
    A Choice control is used to select one of a list of strings. Unlike a
    wx.ListBox, only the selection is visible until the user pulls down
    the menu of choices.

    Events

    EVT_CHOICE  Sent when an item in the list is selected.
    '''
    def __init__(self,
                 parent,
                 id=wx.ID_ANY,
                 pos=wx.DefaultPosition,
                 size=wx.DefaultSize,
                 choices=wx.EmptyList,
                 style=0,
                 validator=wx.DefaultValidator,
                 name=wx.ChoiceNameStr):
        '''
        Create and show a Choice control
        '''
        theClass = 'Choice'

        # Capture initial caller parametsrs before they are changed
        self.caller_parent = parent
        self.caller_id = id
        self.caller_choices = choices
        self.caller_pos = pos
        self.caller_size = size
        self.caller_style = style
        self.caller_validator = validator
        self.caller_name = name

        # TBD - St default label
        label = None

        wx.RegisterFirstCallerClassName(self, theClass)

        ControlWithItems.__init__(self)

        self.tsBeginClassRegistration(theClass, id)

        if DEBUG:
            self.logger.debug('               self: %s' % self)
            self.logger.debug('             parent: %s' % parent)
            self.logger.debug('                 id: %s' % self.ts_Id)
            self.logger.debug('         AssignedId: %s' % self.ts_AssignedId)
            self.logger.debug('            choices: %s' % choices)
            self.logger.debug('                pos: %s' % str(pos))
            self.logger.debug('               size: %s' % str(size))
            self.logger.debug('              style: 0x%X' % style)
            self.logger.debug('               name: %s' % name)

        self.ts_Default = False

        self.ts_Name = name
        self.ts_Parent = parent

        theTopLevelClass = self
        self.SetTopLevelAncestor(theTopLevelClass)

        if parent is None:
            self.ts_GrandParent = None
        else:
            self.ts_GrandParent = parent.Parent

        if False:
            # Leaves artifacts of different color than parent background.
            self.ts_BackgroundColour = wx.ThemeToUse[
                'BackgroundColour'].lower()
            self.ts_ForegroundColour = wx.ThemeToUse[
                'ForegroundColour'].lower()
        else:
            self.ts_BackgroundColour = self.Parent.ts_BackgroundColour
            self.ts_ForegroundColour = self.Parent.ts_ForegroundColour

        self.ts_Style = style

        self.ts_ButtonText = label
        self.ts_Label = label
        self.ts_Validator = validator
        self.ts_Value = False

        myRect, myClientRect = self.tsButtonLayout(
            parent, pos, size, style, name)
        self.ts_Rect = myRect
        self.ts_ClientRect = myClientRect

        self.tsEndClassRegistration(theClass)

    def Create(self,
               parent,
               id=wx.ID_ANY,
               pos=wx.DefaultPosition,
               size=wx.DefaultSize,
               choices=wx.EmptyList,
               style=0,
               validator=wx.DefaultValidator,
               name=wx.ChoiceNameStr,
               pixels=True):
        '''
        Actually create the GUI Choice control for 2-phase creation
        '''
        myRect, myClientRect = self.tsButtonLayout(
            parent, pos, size, style, name)
        self.ts_Rect = myRect
        self.ts_ClientRect = myClientRect

        self.ts_Handle = self.tsCursesNewWindow(myRect, pixels=pixels)

        return (self.ts_Handle is not None)

    @staticmethod
    def GetClassDefaultAttributes(variant=wx.WINDOW_VARIANT_NORMAL):
        '''
        Get the default attributes for this class. This is useful if you want
        to use the same font or colour in your own control as in a standard
        control -- which is a much better idea than hard coding specific
        colours or fonts which might look completely out of place on the
        user system, especially if it uses themes.

        The variant parameter is only relevant under Mac currently and is
        ignored under other platforms. Under Mac, it will change the size of
        the returned font. See wx.Window.SetWindowVariant for more about this.
        '''
        pass

    def GetCurrentSelection(self):
        '''
        Unlike GetSelection which only returns the accepted selection value,
        i.e. the selection in the control once the user closes the dropdown
        list, this function returns the current selection. That is, while the
        dropdown list is shown, it returns the currently selected item in it.
        When it is not shown, its result is the same as for the other function.
        '''
        return (int)

    #-----------------------------------------------------------------------
    # Begin tsWx API Extensions

    # End tsWx API Extensions
    #-----------------------------------------------------------------------

    CurrentSelection = property(GetCurrentSelection)

#---------------------------------------------------------------------------
 
if __name__ == '__main__':

    print(__header__)
